### PR TITLE
test coverage

### LINKED ISSUE #78: [router] Config Unit Test
```describe("Config", () => {
  describe.skip("getEnvConfig", () => {
    it("should read config from default filepath", () => {});
    it("should read config from specified filepath", () => {});
    it("should read config config env var", () => {});
    it("should read individual env vars", () => {});
    it("should preserve the order of specificity (individual env vars > env var config > specified config path > default config path", () => {});
  });
  describe.skip("getConfig", () => {
    it("should getEnvConfig", () => {});
    it("should return from cache", () => {});
  });
});```

### LINKED ISSUE #79: [router] Contract Unit test
```describe("Contract", () => {
  describe.skip("class TransactionManager", () => {
    it("happy: constructor", () => {});
    it("happy: constructor with optional config param", () => {});

    describe.skip("prepare", () => {
      it("should error if chainId is not supported in config", () => {});

      describe("should error if function encoding fails", () => {
        it("invalid user", () => {});
        it("invalid router", () => {});
        it("invalid sendingChainId", () => {});
        it("invalid sendingAssetId", () => {});
        it("invalid receivingChainId", () => {});
        it("invalid receivingAssetId", () => {});
        it("invalid sendingChainFallback", () => {});
        it("invalid callTo", () => {});
        // receivingAddress
        // callDataHash
        // transactionId
        // amount
        // expiry
        // encryptedCallData
        // encodedBid
        // bidSignature
      });

      it("should error if transaction fails", () => {});
      it("happy case: prepare erc20", () => {});
      it("happy case: prepare native", () => {});
    });

    describe.skip("fulfill", () => {
      it("should error if chainId is not supported in config", () => {});
      describe("should error if function encoding fails", () => {});
      it("should error if transaction fails", () => {});
      it("happy case: fulfill erc20", () => {});
      it("happy case: fulfill native", () => {});
    });
    describe.skip("cancel", () => {
      it("should error if chainId is not supported in config", () => {});
      describe("should error if function encoding fails", () => {});
      it("should error if transaction fails", () => {});
      it("happy case: cancel erc20", () => {});
      it("happy case: cancel native", () => {});
    });
    describe.skip("removeLiquidity", () => {
      it("should error if chainId is not supported in config", () => {});
      describe("should error if function encoding fails", () => {});
      it("should error if transaction fails", () => {});
      it("happy case: removeLiquidity erc20", () => {});
      it("happy case: removeLiquidity native", () => {});
    });
    describe.skip("getRouterBalance", () => {
      it("should error if chainId is not supported in config", () => {});
      describe("should error if function encoding fails", () => {});
      it("should error if read transaction fails", () => {});
      it("happy case: getRouterBalance erc20", () => {});
      it("happy case: getRouterBalance native", () => {});
    });
  });
});
```

### LINKED ISSUE #80: [router] handler unit test

### LINKED ISSUE #81: [router] subgraph unit test

### LINKED ISSUE #82: [sdk] sdk unit test

### LINKED ISSUE #83: [sdk] subgraph unit test

### LINKED ISSUE #84: [sdk] transaction manager unit test

### LINKED ISSUE #104: [router] Handle error where tx is already prepared
Error is P:015.

Blocked by #102

### LINKED ISSUE #117: [sdk] Validation of method input parameters

### LINKED ISSUE #118: [router] Validation of method input parameters

### PR DESCRIPTION
**Pull request by sanchaymittal**
closes #78 
closes #79
closes #80 
closes #81 
closes #82 
closes #83 
closes #84 
 
Fixes #104 
closes #117
closes #118

### CODE REVIEW

File: packages/router/src/config.ts [hunk 1]
```diff
@@ -153,7 +152,11 @@ export const getEnvConfig = (): NxtpRouterConfig => {
       : configJson.chainConfig
       ? configJson.chainConfig
       : configFile.chainConfig,
-    swapPools: process.env.NXTP_SWAP_POOLS || configJson.swapPools || configFile.swapPools,
+    swapPools: process.env.NXTP_SWAP_POOLS
+      ? JSON.parse(process.env.NXTP_SWAP_POOLS)
```
Comment (rhlsthrm) [inline packages/router/src/config.ts:156]: Nice catch.

File: packages/router/src/config.ts [hunk 2]
```diff
@@ -187,7 +190,7 @@ export const getEnvConfig = (): NxtpRouterConfig => {
   const valid = validate(nxtpConfig);
 
   if (!valid) {
-    console.error(`Invalid config: ${JSON.stringify(nxtpConfig, null, 2)}`);
+    console.error(`Invalid config: ${JSON.stringify({ ...nxtpConfig, mnemonic: "********" }, null, 2)}`);
```
Comment (rhlsthrm) [inline packages/router/src/config.ts:193]: Nice.

File: packages/sdk/src/sdk.ts [hunk 3]
```diff
@@ -204,13 +265,37 @@ export class NxtpSdkError extends NxtpError {
...
   ) {
     super(message, context, TransactionManagerError.type);
   }
 }
 
+/**
+ * This is only here to make it easier for sinon mocks to happen in the tests. Otherwise, this is a very dumb thing.
+ *
+ */
+export const sdkSignFulfillTransactionPayload = async (
```
Comment (rhlsthrm) [inline packages/sdk/src/sdk.ts:280]: We should probably keep it as the same name as the other function and do something like 
```ts
import { signFulfillTransactionPayload as _signFulfillTransactionPayload } from "@connext/nxtp-utils"
```

**Review by rhlsthrm — APPROVED** [review]
Nice work!

File: packages/sdk/src/sdk.ts (hunk 3 above)
Comment (sanchaymittal) [reply packages/sdk/src/sdk.ts:280]: yeah make sense

Comment (sanchaymittal) [reply packages/sdk/src/sdk.ts]: done

**Pull request merged**